### PR TITLE
Feature completed

### PR DESCRIPTION
**Pull request by MasterRadule**

### CODE REVIEW

File: Backend/src/main/java/ktsnvt/tim1/DTOs/EventDTO.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        this.reservationDeadlineDays = event.getReservationDeadlineDays();
+        this.cancelled = event.getCancelled();
+        this.maxReservationsPerUser = event.getMaxReservationsPerUser();
+        this.eventDays = event.getEventDays().stream().map(EventDayDTO::new)
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    public Event convertToEntity() throws ParseException {
+        Event e = new Event();
+        e.setId(this.id);
```
Comment (aleksaleks123) [inline Backend/src/main/java/ktsnvt/tim1/DTOs/EventDTO.java]: Id needs to be null

File: Backend/src/main/java/ktsnvt/tim1/DTOs/EventDTO.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    @Positive
+    private Integer reservationDeadlineDays;
+
+    @Positive
+    private Integer maxReservationsPerUser;
+
+    @Valid
+    @NotNull(message = "Event days must be specified")
+    @NotEmpty(message = "Event days must be specified")
+    private ArrayList<EventDayDTO> eventDays;
```
Comment (draganculibrk9) [inline Backend/src/main/java/ktsnvt/tim1/DTOs/EventDTO.java]: @NotEmpty checks if reference is null and also if collection is empty, remove @NotNull as it is not necessary

File: Backend/src/main/java/ktsnvt/tim1/DTOs/EventDayDTO.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public EventDayDTO(EventDay eventDay) {
+        this.id = eventDay.getId();
+        this.date = formatter.format(eventDay.getDate());
+    }
+
+    public EventDay convertToEntity() throws ParseException {
+        EventDay ed = new EventDay();
+        ed.setId(this.id);
```
Comment (draganculibrk9) [inline Backend/src/main/java/ktsnvt/tim1/DTOs/EventDayDTO.java]: EventDay id should be set to null

File: Backend/src/main/java/ktsnvt/tim1/controllers/EventController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import javax.validation.Valid;
+import java.text.ParseException;
+
+@RestController
+@RequestMapping("/events")
+public class EventController {
+    @Autowired
+    private EventService eventService;
+
```
Comment (draganculibrk9) [inline Backend/src/main/java/ktsnvt/tim1/controllers/EventController.java]: getEvents method is missing

**Pull request merged**